core/server/lib/image/manipulator.js

```diff
@@ -8,15 +9,22 @@ const common = require('../common');
  * https://github.com/lovell/sharp/issues/1360.
  */
 const process = (options = {}) => {
-    let sharp, img;
+    let sharp, img, init, originalData, originalSize;
 
     try {
         sharp = require('sharp');
 
         // @NOTE: workaround for Windows as libvips keeps a reference to the input file
         //        which makes it impossible to fs.unlink() it on cleanup stage
         sharp.cache(false);
-        img = sharp(options.in);
+
+        init = fs.readFile(options.in)
+            .then((data) => {
+                originalData = data;
+
+                // @NOTE: have to user constructor with Buffer for sharp to be able to expose size property
+                img = sharp(data);
+            });
```

Comment:
This becomes a little hard to read, because the try/catch is now used for sync and async code 🤔

Reply:
100% agree. Have refactored it into a more readable piece of code :)